src/test/java/org/opensearch/search/asynchronous/bwc/AsyncSearchBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,357 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.search.asynchronous.bwc;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.search.asynchronous.context.active.AsynchronousSearchActiveStore;
+import org.opensearch.search.asynchronous.context.state.AsynchronousSearchState;
+import org.opensearch.search.asynchronous.request.DeleteAsynchronousSearchRequest;
+import org.opensearch.search.asynchronous.request.GetAsynchronousSearchRequest;
+import org.opensearch.search.asynchronous.request.SubmitAsynchronousSearchRequest;
+import org.opensearch.search.asynchronous.response.AsynchronousSearchResponse;
+import org.opensearch.search.asynchronous.restIT.AsynchronousSearchRestTestCase;
+import org.opensearch.search.asynchronous.service.AsynchronousSearchService;
+import org.opensearch.search.asynchronous.settings.LegacyOpendistroAsynchronousSearchSettings;
+import org.opensearch.search.asynchronous.utils.RestTestUtils;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.Map;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Arrays;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class AsyncSearchBackwardsCompatibilityIT extends AsynchronousSearchRestTestCase {
+    private static final ClusterType CLUSTER_TYPE =
+            ClusterType.parse(System.getProperty("tests.rest.bwcsuite_cluster"));
+    private static final String CLUSTER_NAME = System.getProperty("tests.clustername");
+
+    public void testBackwardsCompatibility() throws Exception {
+        String uri = getUri();
+        Map<String, Map<String, Object>> responseMap =
+                (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<Object> pluginNames =
+                    plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
+            switch (CLUSTER_TYPE) {
+                case OLD:
+                case MIXED:
+                case UPGRADED:
+                    testAsyncSearchAndSettingsApi(true);
```

Comment:
Is `shouldUseLegacyApi` true even for an upgraded cluster?

Reply:
I initially wrote with a different case where legacy will be used only for old and mixed. But changed it to check if legacy APIs will work for all 3 cases.

Is there any recommendation for upgraded cluster ? should we use the current APIs for upgraded instead of legacy ?